groot/routes.py

```diff
@@ -1,17 +1,52 @@
 from flask import Flask, Response, render_template, request, redirect, url_for, flash
 from run import app, db
-
+from groot.models import *
+import hashlib
 
 
 @app.route("/")
 @app.route('/login', methods=['POST', 'GET'])
 def login():
-    return render_template('login.html')
-        
+    if request.method == 'POST':
+        try:
+            email = request.form.get('email')
+            password = request.form.get('password')    
+            user = User.query.filter_by(email=email).first()
+            if user:
+                hash_pw = hashlib.sha1(password.encode('utf-8')).hexdigest().upper()
+                if user.password == hash_pw:
+                    return redirect(url_for('dashboard'))  
+                else:
+                    flash('Inccorect password! Please try again.', 'danger')
+            else:      
+                flash('Inccorect email! Please try again.', 'danger')    
+            return redirect(url_for('login'))  
+        except:
+            return 'Should return 404 error page'
```

Comment:
There is an easy way to return errors like 404 with flask, check out flask's "abort" module.

Reply:
cool i will check it out!